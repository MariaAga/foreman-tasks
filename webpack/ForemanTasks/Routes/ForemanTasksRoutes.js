import React from 'react';
import { TasksIndexPage } from '../Components/TasksTable/TasksIndexPage';
import { SubTasksPage } from '../Components/TasksTable/SubTasksPage';
import ShowTask from './ShowTask';

const ForemanTasksRoutes = {
  indexTasks: {
    path: '/foreman_tasks/tasks',
    exact: true,
    render: props => <TasksIndexPage {...props} />,
  },
  subTasks: {
    path: '/foreman_tasks/tasks/:id/sub_tasks',
    exact: true,
    render: props => <SubTasksPage {...props} />,
  },
  showTask: {
    path: '/foreman_tasks/ex_tasks/:id',
    render: props => <ShowTask {...props} />,
  },
};

export default ForemanTasksRoutes;
