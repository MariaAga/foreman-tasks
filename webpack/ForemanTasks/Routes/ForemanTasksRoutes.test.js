import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import ForemanTasksRoutes from './ForemanTasksRoutes';

describe('ForemanTasksRoutes', () => {
  it('should create routes', () => {
    Object.entries(ForemanTasksRoutes).forEach(([key, Route]) => {
      const component = shallow(
        <Route.render
          history={{
            push: jest.fn(),
          }}
          some="props"
        />
      );
      Route.renderResult = toJson(component);
    });

    expect(ForemanTasksRoutes).toMatchSnapshot();
  });
});
