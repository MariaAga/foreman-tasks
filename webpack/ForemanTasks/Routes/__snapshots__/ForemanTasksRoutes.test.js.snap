// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForemanTasksRoutes should create routes 1`] = `
Object {
  "indexTasks": Object {
    "exact": true,
    "path": "/foreman_tasks/tasks",
    "render": [Function],
    "renderResult": <TasksIndexPage
      some="props"
    />,
  },
  "showTask": Object {
    "path": "/foreman_tasks/ex_tasks/:id",
    "render": [Function],
    "renderResult": <ShowTask
      some="props"
    />,
  },
  "subTasks": Object {
    "exact": true,
    "path": "/foreman_tasks/tasks/:id/sub_tasks",
    "render": [Function],
    "renderResult": <SubTasksPage
      match={
        Object {
          "params": Object {},
        }
      }
      some="props"
    />,
  },
}
`;
